lighthouse-core/audits/dobetterweb/has-inspector-issues.js

```diff
@@ -0,0 +1,192 @@
+      requestUrls.add(requestUrl);
+    }
+    return {
+      issueType: 'Mixed Content',
+      subItems: {
+        type: 'subitems',
+        items: Array.from(requestUrls).map(url => ({url})),
+      },
+    };
+  }
+
+  /**
+   * @param {Array<LH.Crdp.Audits.SameSiteCookieIssueDetails>} sameSiteCookieIssues
+   * @return {LH.Audit.Details.TableItem}
+   */
+  static getSameSiteCookieRow(sameSiteCookieIssues) {
+    const requestUrls = new Set();
+    for (const issue of sameSiteCookieIssues) {
+      const requestUrl = (issue.request && issue.request.url) || issue.cookieUrl;
+      if (requestUrl) {
+        requestUrls.add(requestUrl);
+      }
+    }
+    return {
+      issueType: 'SameSite Cookie',
+      subItems: {
+        type: 'subitems',
+        items: Array.from(requestUrls).map(url => {
+          return {
+            url,
+          };
+        }),
+      },
+    };
+  }
+
+  /**
+   * @param {Array<LH.Crdp.Audits.BlockedByResponseIssueDetails>} blockedByResponseIssues
+   * @return {LH.Audit.Details.TableItem}
+   */
+  static getBlockedByResponseRow(blockedByResponseIssues) {
+    const requestUrls = new Set();
+    for (const issue of blockedByResponseIssues) {
+      const requestUrl = issue.request && issue.request.url;
+      if (requestUrl) {
+        requestUrls.add(requestUrl);
+      }
+    }
+    return {
+      issueType: str_(UIStrings.issueTypeBlockedByResponse),
+      subItems: {
+        type: 'subitems',
+        items: Array.from(requestUrls).map(url => {
+          return {
+            url,
+          };
+        }),
+      },
+    };
+  }
+
+  /**
+   * @param {Array<LH.Crdp.Audits.ContentSecurityPolicyIssueDetails>} cspIssues
+   * @return {LH.Audit.Details.TableItem}
+   */
+  static getContentSecurityPolicyRow(cspIssues) {
+    const requestUrls = new Set();
+    for (const issue of cspIssues) {
+      const requestUrl = issue.blockedURL;
+      if (requestUrl) {
+        requestUrls.add(requestUrl);
+      }
+    }
+    return {
+      issueType: 'Content Security Policy',
+      subItems: {
+        type: 'subitems',
+        items: Array.from(requestUrls).map(url => {
+          return {
+            url,
+          };
+        }),
+      },
+    };
+  }
+
+  /**
+   * @param {LH.Artifacts} artifacts
+   * @return {LH.Audit.Product}
+   */
+  static audit(artifacts) {
+    /** @type {LH.Audit.Details.Table['headings']} */
+    const headings = [
+      /* eslint-disable max-len */
+      {key: 'issueType', itemType: 'text', subItemsHeading: {key: 'url', itemType: 'url'}, text: str_(UIStrings.columnIssueType)},
+      /* eslint-enable max-len */
+    ];
+
+    const issues = artifacts.InspectorIssues;
+    /** @type LH.Audit.Details.TableItem[] */
+    const items = [];
+
+    if (issues.mixedContent.length) {
+      items.push(this.getMixedContentRow(issues.mixedContent));
+    }
+    if (issues.sameSiteCookies.length) {
+      items.push(this.getSameSiteCookieRow(issues.sameSiteCookies));
+    }
```

Comment:
Sorry, was just thinking about the filtering of issues in the audit and realized a corner case we'll need to handle.

In theory, this checks that there are `sameSiteCookies` issues before calling `this.getSameSiteCookieRow()`, which means it makes sense to just push the result. However, it's possible there are elements in `issues.sameSiteCookies`, but then they'll be filtered out on L81, resulting in this pushing a `'SameSite Cookie'` row with zero subitems.

I assume that's not what we want? (`heavyAds` doesn't have subitems, but it seems like we should only show `'SameSite Cookie'` or `'Content Security Policy'`, etc if there's an error the user can be told to fix)

It might make sense to change to a different model for these, then, e.g. have `getSameSiteCookieRow` return `null` if `requestUrls.size === 0`, then switch this around to something like

```js
const sameSiteCookieItem = this.getSameSiteCookieRow(issues.sameSiteCookies);
if (sameSiteCookieItem) {
  items.push(sameSiteCookieItem);
}
```
(or some variation on this) here and for the other issue types.

Reply:
My take on it was even if we didn't get the requested Url for whatever reason, we'd still want to show the overarching issue type if for nothing else but to let them know that they exist in the Issue panel. So not having any associated urls shouldn't necessarily mean we ignore the issues. Heavy Ads is kind of in its own boat as it never has an associated request url.